### PR TITLE
Added ADR for additional checksums

### LINKED ISSUE #513: Ensure that checksums are appropriately configured in S3 for preservation
Ensure that file checksums are being recorded and available via inventory retrieval for all PDC Describe buckets.

## Resources: 

* https://aws.amazon.com/blogs/aws/new-additional-checksum-algorithms-for-amazon-s3/

### PR DESCRIPTION
**Pull request by kelynch**
Closes #513

### CODE REVIEW

File: architecture-decisions/0009-Additional-checksums.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+Call: 
+
+```bash
+aws s3api head-object --bucket $BUCKET --key $OBJECT_KEY --output json --checksum-mode ENABLED
+```
+
+Response:
+
+```bash
```
[the remark this answers is not in the document]
Comment (carolyncole) [reply architecture-decisions/0009-Additional-checksums.md:34]: see the awscli response below matches the ruby client.
```
colec in ~/projects/pdc_describe (i1153-approve-require-doi) > aws s3api head-object --bucket $BUCKET --key $OBJECT_KEY --output json --checksum-mode ENABLED
{
    "AcceptRanges": "bytes",
    "LastModified": "2023-05-24T15:27:34+00:00",
    "ContentLength": 943363,
    "ETag": "\"9a9166e3ba6890cd43fe2ca5b77af496\"",
    "VersionId": "gxl2EbqsuApVtKlYv1cEkmQZtuEDk7.b",
    "ContentType": "image/png",
    "ServerSideEncryption": "AES256",
    "Metadata": {}
}
```

**Review by carolyncole — CHANGES_REQUESTED** [review]
I'm not sure if we should be recommending turning on checksums on upload, but the information you have here seems good.  Should it include the fact that none of our objects will have a SHA256, so calling the head_object will not create or retrieve one?

File: architecture-decisions/0009-Additional-checksums.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+This research was done as part of the [PDC Describe preservation plan](https://docs.google.com/document/d/1PG2yQCdOAgKuEcXpX578ED6VCahmel1PRchOpi1hYwQ/edit#).
+
+[Additional checksums](https://aws.amazon.com/blogs/aws/new-additional-checksum-algorithms-for-amazon-s3/) in addition to MD5 can be recorded for objects in S3 if enabled at upload.  There are suggested methods for retroactively enabling additional checksums for objects already uploaded to S3, and a lambda-based strategy for enabling them for all newly uploaded objects in [this article from AWS](https://aws.amazon.com/blogs/storage/enabling-and-validating-additional-checksums-on-existing-objects-in-amazon-s3/).
+
+Based on preliminary exploration, additional checksums do not appear to be available via the aws-sdk Rubygem’s code, despite the fact that there are instance methods referencing them, example:
+
+Call and output on an object with the SHA-256 checksum set:
+
+```ruby
+pry(main)> object.checksum_sha256
```
[the remark this answers is not in the document]
Comment (kelynch) [reply architecture-decisions/0009-Additional-checksums.md:20]: Interesting, thanks for this.  Confusingly, I still got "nil" back on an object where I had enabled the checksum at upload and could get at that value through the API directly, and in the S3 web interface ("kate_test.txt" in the staging post-curation bucket).  I'm wondering what I may have missed.

[the remark this answers is not in the document]
Comment (carolyncole) [reply architecture-decisions/0009-Additional-checksums.md:20]: Interesting, I just did a quick check was able to get them back with `object = s3_client.head_object({bucket: "pdc-describe-prod-precuration", key: "README.txt", checksum_mode: "ENABLED"})` It then had a sha256 checksum.

**Review by carolyncole — APPROVED** [review]

**Pull request merged**